src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlMathTranslator.cs

```diff
@@ -19,117 +20,217 @@
 // AND FITNESS FOR A PARTICULAR PURPOSE. THE SOFTWARE PROVIDED HEREUNDER IS
 // ON AN "AS IS" BASIS, AND THE NPGSQL DEVELOPMENT TEAM HAS NO OBLIGATIONS
 // TO PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+
 #endregion
 
 using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Reflection;
 using System.Linq.Expressions;
+using JetBrains.Annotations;
 using Microsoft.EntityFrameworkCore.Query.Expressions;
 using Microsoft.EntityFrameworkCore.Query.ExpressionTranslators;
-using Npgsql.EntityFrameworkCore.PostgreSQL.Utilities;
+using Npgsql.EntityFrameworkCore.PostgreSQL.Query.Expressions.Internal;
 
 namespace Npgsql.EntityFrameworkCore.PostgreSQL.Query.ExpressionTranslators.Internal
 {
+    /// <summary>
+    /// Provides translation services for static <see cref="Math"/> methods..
+    /// </summary>
+    /// <remarks>
+    /// See:
+    ///   - https://www.postgresql.org/docs/current/static/functions-math.html
+    ///   - https://www.postgresql.org/docs/current/static/functions-conditional.html#FUNCTIONS-GREATEST-LEAST
+    /// </remarks>
     public class NpgsqlMathTranslator : IMethodCallTranslator
     {
-        static readonly Dictionary<MethodInfo, string> _supportedMethodTranslations = new Dictionary<MethodInfo, string>
-        {
-            { typeof(Math).GetRuntimeMethod(nameof(Math.Abs), new[] { typeof(decimal) }), "ABS" },
```

Comment:
Since we're in a translator for methods from a single static class, it seems like overkill to go through a lookup on the `MethodInfo`. The lookup is terse...but a constant-based switch statement feels more natural, since its modeling a fixed set of cases directly.

Also, not that we need to avoid reflection, but if we can avoid caching a bunch of `MethodInfo` without sacrificing perf/clarity, then why not?

Reply:
I don't have extremely strong feelings about this either way, but I don't really see a reason to avoid caching `MethodInfo`s in a table... The translator is a singleton object so this has no impact...

Note also that a `MethodInfo` lookup table is how the [`SqlServer` translator works](https://github.com/aspnet/EntityFrameworkCore/blob/10a8ddf1d7ea5aecbd69336c4e57122cb5a68bc0/src/EFCore.SqlServer/Query/ExpressionTranslators/Internal/SqlServerMathTranslator.cs#L21). Not that parity here is that important, but that's the choice they made.

I'm OK with keeping your new way of doing it. As a general rule I'd avoid making this kind of refactor unless there's a good reason (added value), simply because any change introduces a risk of error, potential merge conflicts, etc... (aka if it works don't fix it :)).